Source/DafnyStandardLibraries/src/DafnyStdLibs/Functions.dfy

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************************
+ *  Copyright by the contributors to the Dafny Project
+ *  SPDX-License-Identifier: MIT 
+ *******************************************************************************/
+/**
+A collection of predicates that establish whether commonly known properties on functions, 
+such as injectivity and commutativity, hold.
+These properties can only be applied to functions on value types.
+*/
+module DafnyStdLibs.Function {
+
+  ghost predicate Injective<X(!new), Y>(f: X --> Y)
```

Comment:
Just FYI, the variants with `(!new)` are more convenient to use, but I expect we'll also want variants where the domain is explicitly passed in as an `iset`, so that they can be used with reference types too. 

I don't think that is blocking adding these versions now, only we might want to have a plan for how to name the variants intuitively. Perhaps this:

```dafny
ghost predicate InjectiveOn<X, Y>(f: X --> Y, ghost iset<X> domain)
  reads f.reads
  requires forall x <- domain :: f.requires(x)
  {
    forall x1 <- domain, x2 <- domain :: f(x1) == f(x2) ==> x1 == x2
  }
```

Or `InjectiveOver`?